### PR TITLE
Handle private key policy errors for the web UI

### LINKED ISSUE #16168: Hardware key WebUI registration
As noted in the [hardware key integration RFD](https://github.com/gravitational/teleport/pull/14774), hardware key login cannot be supported in the WebUI because it requires local access to your hardware key (yubikey).

However, with Teleport, registration and password resetting can only be done on the WebUI. These functions should remain possible even if a user is required to login with a hardware key.

Hardware key login is currently a WIP, but the core functionality is complete in https://github.com/gravitational/teleport/pull/15874. With this PR, registration and password reset succeeds, but ends with an error in the WebUI:

register

The WebUI should be updated to catch `private key policy not met` errors, and redirect users to an appropriate page. This could be a page which says something like "This user requires a hardware key, such as yubikey, to login. This functionality is not supported in the Teleport WebUI. Please use `tsh` or Teleport Connect to login as this user instead."

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/16168

#### Description
Please see https://github.com/gravitational/webapps/pull/1298, for UI and copy review. I've tried to target all edge cases, let me know if you folks come up with more.

- When user registers or resets, when running into
policy errors, we don't send back an error, but instead
a 200 (to indicate user has successfully registered/resetted)
and a flag to determine if policy was enabled
- Send back policy configurations from cluster config,
this will allow the web UI login page to just display
redirection messages without allowing any login attempts
- For role configured policy, login attempts will be
required and a specific error will be returned so that
the UI can then better redirect the user

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

File: lib/auth/password.go [hunk 1]
```diff
@@ -62,6 +63,17 @@ func (s *Server) ChangeUserAuthentication(ctx context.Context, req *proto.Change
 
 	webSession, err := s.createUserWebSession(ctx, user)
 	if err != nil {
+		if yes := keys.IsPrivateKeyPolicyError(err); yes {
```
Comment (marcoandredinis) [inline lib/auth/password.go:66]:
```suggestion
		if keys.IsPrivateKeyPolicyError(err) {
```

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -1861,6 +1869,21 @@ func (h *Handler) changeUserAuthentication(w http.ResponseWriter, r *http.Reques
...
+	if res.PrivateKeyPolicyEnabled {
+		if res.GetRecovery() == nil {
+			return &ui.ChangedUserAuthn{
+				PrivateKeyPolicyEnabled: res.PrivateKeyPolicyEnabled,
+			}, nil
+		}
+		return &ui.ChangedUserAuthn{
+			Recovery: ui.RecoveryCodes{
+				Codes:   res.Recovery.Codes,
+				Created: &res.Recovery.Created,
```
Comment (marcoandredinis) [inline lib/web/apiserver.go:1881]: :nail_care: 

Above we use the method `Recovery()` but here we access the property directly.

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -2036,6 +2061,12 @@ func (h *Handler) mfaLoginFinishSession(w http.ResponseWriter, r *http.Request,
 	clientMeta := clientMetaFromReq(r)
 	session, err := h.auth.AuthenticateWebUser(r.Context(), req, clientMeta)
 	if err != nil {
+		// Since checking for private key policy meant that they passed authn,
+		// return policy error as is to help direct user.
+		if yes := keys.IsPrivateKeyPolicyError(err); yes {
```
Comment (marcoandredinis) [inline lib/web/apiserver.go:2066]:
```suggestion
		if keys.IsPrivateKeyPolicyError(err) {
```

**Review by marcoandredinis — APPROVED** [review]
LGTM

**Review by Joerger — APPROVED** [review]
Can we backport this to v11?

**Pull request merged**